### DISCUSSION TITLE
PricingStream is not working

### DISCUSSION THREAD

**Discussion by msh855** [hootnot/oanda-api-v20#216] [General]
Hi, 

This example from your documentation is not working: 
```

>>> r = pricing.PricingStream(accountID=accountID, params=params)
>>> rv = api.request(r)
>>> maxrecs = 100
>>> for ticks in r:
>>>     print json.dumps(R, indent=4),","
>>>     if maxrecs == 0:
>>>         r.terminate("maxrecs records received")
```

I get this error: `'PricingStream' object is not iterable `

Here what I am doing: 

```
cross = 'USD_JPY'
r = pricing.PricingStream(accountID=account_id, params={"instruments": cross})
rv = api.request(r)
```

**Comment by hootnot** [comment]
Indeed, the request object is not iterable, the result of the client request of the object is

https://github.com/hootnot/oandapyV20-examples/blob/master/src/streaming_prices.py

```
for ticks in rv:   # not r !
```

**Comment by msh855** [comment]
Got it. I think the documentation needs to change as well. Possibly it is a typo, but needs to be fixed.

**Comment by DICAP1** [comment]
Hey everyone,

I've been digging through some of the Python automation built around the Oanda v20 REST API. It's awesome how many clean execution frameworks have been built on top of oandapyV20 over the years, but I think we’ve all run into that same structural wall: B-Book execution toxicity and sudden volatility spikes.

It's an absolute nightmare when an algo functions beautifully in a backtest, but then burns cash live because a retail pool dynamically widens spreads or slips your order during an unannounced liquidity gap.

A few of us have been building an intelligence-driven pre-trade filter called Sovereign (deepalgo.co.uk) to tackle this exactly. It computes real-time Hidden Markov Model (HMM) market regimes and order flow toxicity (VPIN) right at the liquidity layer.

We designed it to slot seamlessly into custom Python strategy files as a fast, non-blocking middleware check right before your code commits an order to Oanda. Something like this:

Python
# Simple pre-flight guard before firing to Oanda
headers = {"X-DeepAlgo-Key": "YOUR_BETA_KEY"}
res = requests.post("https://api.deepalgo.co.uk/v4/approve", json={"instrument": "EUR_USD"}, headers=headers)

if res.json().get("status") == "APPROVED":
    # Safe regime: fire your oandapyV20 OrderCreate endpoint
    api.request(orders.OrderCreate(accountID, data=order_data)) 
else:
    logger.warning("Sovereign triggered a VETO — standing down to protect account equity.")

We’re kicking off our early alpha/beta tracking phase right now and want to get a solid group of independent FX quants to break the endpoints, look at the latency, and see how it alters their risk metrics. We are hand-allocating free beta access keys seeded with a 10,000 credit tracking block so you can play with it in your dev environment without limits.

If you’re running live or demo Oanda scripts and want to grab a key + documentation to mess around with, drop us a line at support@deepalgo.co.uk and we'll get you set up.

Would love to get your raw feedback on the hot-path integration!
